### PR TITLE
Adding startupProbe to airflow services charts

### LINKED ISSUE #33099: Add startupProbe to airflow helm charts
### Description

Introducing a startupProbe onto the airflow services would be useful for slow starting container and most of all it doesn't have side effects.

### Use case/motivation

We have an internal feature where we perform a copy of venv from airflow services to cloud storages which can sometimes take a few minutes. Copying of a venv is a metadata heavy load: https://learn.microsoft.com/en-us/troubleshoot/azure/azure-storage/files-troubleshoot-performance?tabs=linux#cause-2-metadata-or-namespace-heavy-workload.

Introducing a startupProbe onto the airflow services would be useful for slow starting container and most of all it doesn't have side effects.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by amoghrajesh**
We have an internal feature where we perform a copy of venv from airflow services to cloud storages which can sometimes take a few minutes. Copying of a venv is a metadata heavy load: https://learn.microsoft.com/en-us/troubleshoot/azure/azure-storage/files-troubleshoot-performance?tabs=linux#cause-2-metadata-or-namespace-heavy-workload.

Introducing a startupProbe onto the airflow services would be useful for slow starting container and most of all it doesn't have side effects. startupProbe has been added for scheduler and the webserver following the same flow as their respective liveness and readiness probes. A gracious value of 10 mins has been set for startup. This is based on personal experiments where our deployments took 5 - 6 mins for startup roughly, 10 mins allows some head room. Although, these properties are configurable and would be beneficial for users to use them.

closes: #33099 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
@jedcunningham @hussein-awala can you take a look at this PR when you have some time?

File: chart/values.yaml [hunk 1]
```diff
@@ -721,6 +721,14 @@ scheduler:
     failureThreshold: 5
     periodSeconds: 60
     command: ~
+
+  # Wait for at most 10 minutes (60*10s) for the scheduler container to startup.
+  # livenessProbe kicks in after the startup
+  startupProbe:
+    failureThreshold: 60
```
Comment (potiuk) [inline chart/values.yaml]: I thin the defaults are pretty long. If there is an error, the user in default configuraiton will have to waut for 10 minutes before they find out something is wrong. 

Minute as a default should be quite enough IMHO

**Comment by potiuk** [comment]
Yes - I think it'a good idea to add startup probe  (but with lower default time) @jedcunningham @dstandish  WDYT

File: chart/values.yaml (hunk 1 above)
Comment (amoghrajesh) [reply chart/values.yaml]: Makes sense. Let me make the changes for that.

**Review by potiuk — APPROVED** [review]

**Pull request merged**